### ISSUE TITLE
--constructor-args is unsound for constructor(String)

### ISSUE THREAD

**Issue by mattsse** [foundry-rs/foundry#1263]
### Component

Forge

### Have you ensured that all of these are up to date?

- [ ] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

master

### What command(s) is the bug in?

verify* create

### Describe the bug

the `forge create` constructor args:

```console
--constructor-args <constructor_args>...
            constructor args calldata arguments
```

accepts multiple values, like `--constructor-args "ForgeUSD" "FUSD"` (ref book)

the provided arguments get abi encoded before the contract is deployed.

This allows for something like mentioned here:
 https://github.com/gakonst/foundry/issues/852#issuecomment-1094536685

```bash
cast abi-encode "constructor(string)" "geek"
```
```console
0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000046765656b00000000000000000000000000000000000000000000000000000000
```

```bash
forge create --constructor-args 0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000046765656b00000000000000000000000000000000000000000000000000000000
```

in which case, what actually gets deployed is:

```bash
cast abi-encode "constructor(string)" "0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000046765656b00000000000000000000000000000000000000000000000000000000"
```

```
0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000c23078303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303032303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303436373635363536623030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030000000000000000000000000000000000000000000000000000000000000
```
which would be the appropriate input for verification

**Comment by onbjerg** [comment]
Was this fixed in #1198? :thinking:

**Comment by mattsse** [comment]
think we can close this, the help messages should be clear on this now
